blake2/src/impl_hacl/mod.rs

```diff
@@ -0,0 +1,249 @@
+extern crate alloc;
+
+mod blake2b;
+mod blake2s;
+mod error;
+
+pub use blake2b::{Blake2b, Blake2bBuilder};
+pub use blake2s::{Blake2s, Blake2sBuilder};
+pub use error::Error;
+
+/// Type that holds the constants in case both key length and digest length are known at compile
+/// time.
+pub struct ConstsStaticKeyStaticOut<const KEY_LEN: usize, const OUT_LEN: usize>;
+
+/// Type that holds the constant in case just the key length is known at compile time.
+pub struct ConstsStaticKey<const KEY_LEN: usize>;
+
+/// Type that holds the constant in case just the digest length is known at compile time.
+pub struct ConstsStaticOut<const OUT_LEN: usize>;
+
+/// Type that holds the constant in case neither the key length nor the digest length is known at
+/// compile time.
+pub struct ConstsDynamic;
+
+#[cfg(test)]
+mod test {
```

Comment:
These tests should be for the crate, not only the hacl implementation. Let's move them up to the lib? Er even out as integration tests, we don't seem to use internals here.